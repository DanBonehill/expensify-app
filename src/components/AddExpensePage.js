import React from "react";

const AddExpensePage = () => (
    <div>
        This is from my Add Expense
    </div>
);

export default AddExpensePage