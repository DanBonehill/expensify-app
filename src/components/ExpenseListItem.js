import React from 'react';

export default ({description, amount, createdAt}) => (
    <div>
        <ul>
            <li>{description}</li>
            <li>{amount}</li>
            <li>{createdAt}</li>
        </ul>
    </div>
)