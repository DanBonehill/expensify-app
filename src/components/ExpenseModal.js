import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement('#app');

const ExpenseModal = (props) => {
    return (
        <Modal
            isOpen={props.isOpen}
            onRequestClose={props.closeModal}
            closeTimeoutMS={150}
            ariaHideApp={ !process.env.NODE_ENV === 'test' }
            className="modal"
        >
            <h3 className="modal__title">Are you sure?</h3>
            <div>
                <button onClick={props.onDeleteClick} className="button button--delete">Remove</button>
                <button onClick={props.closeModal} className="button button--link">Cancel</button>
            </div>
        </Modal>
    )
};

export default ExpenseModal