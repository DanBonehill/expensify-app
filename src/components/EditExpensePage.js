import React from "react";
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startEditExpense, startRemoveExpense} from "../actions/expenses";
import ExpenseModal from './ExpenseModal';

export class EditExpensePage extends React.Component {
    state = {
        isOpen: false
    };
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onClick = () => {
        this.setState(() => ({
            isOpen: true
        }));
    };
    onDeleteClick = () => {
        this.props.startRemoveExpense({id: this.props.expense.id});
        this.props.history.push('/');
    };
    closeModal = () => {
        this.setState(() => ({
            isOpen: false
        }))
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onClick}>Remove Expense</button>
                    <ExpenseModal
                        isOpen={this.state.isOpen}
                        closeModal={this.closeModal}
                        onDeleteClick={this.onDeleteClick}
                    />
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)