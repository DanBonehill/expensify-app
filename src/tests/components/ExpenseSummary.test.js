import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import {ExpensesSummary, ExpenseSummary} from '../../components/ExpensesSummary';

describe('ExpenseSummary snapshot', () => {
    test('should render correctly with one expense', () => {
        const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={20000}/>);
        expect(wrapper).toMatchSnapshot()
    });

    test('should render correctly with two expenses', () => {
        const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={215000}/>);
        expect(wrapper).toMatchSnapshot()
    })
});