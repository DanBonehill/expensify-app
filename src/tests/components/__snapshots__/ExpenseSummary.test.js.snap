// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseSummary snapshot should render correctly with one expense 1`] = `
<div>
  <h5>
    Viewing 
    1
     
    expense
     totaling 
    £200.00
  </h5>
</div>
`;

exports[`ExpenseSummary snapshot should render correctly with two expenses 1`] = `
<div>
  <h5>
    Viewing 
    2
     
    expenses
     totaling 
    £2,150.00
  </h5>
</div>
`;
